cardano-db-sync/src/Cardano/DbSync/Plugin/Default/Byron/Rollback.hs

```diff
@@ -37,18 +36,18 @@ rollbackToPoint trce point =
   where
     action :: MonadIO m => SlotNo -> Byron.HeaderHash -> ExceptT DbSyncNodeError (ReaderT SqlBackend m) ()
     action slot hash = do
-        blk <- liftLookupFail "rollbackToPoint" $ DB.queryMainBlock (Byron.unHeaderHash hash)
-        case DB.blockSlotNo blk of
-          Nothing -> dbSyncNodeError "rollbackToPoint: slot number is Nothing"
-          Just slotNo -> do
-            if slotNo <= unSlotNo slot
-              then liftIO . logInfo trce $ mconcat
-                            [ "Byron: No rollback required: db tip slot is ", textShow slotNo
-                            , " ledger tip slot is ", textShow (unSlotNo slot)
-                            ]
-              else do
-                liftIO . logInfo trce $ Text.concat
-                            [ "Rollbacking to slot ", textShow (unSlotNo slot)
-                            , ", hash ", Byron.renderAbstractHash hash
-                            ]
-                void . lift $ DB.deleteCascadeSlotNo slotNo
+        slotNo <- lift DB.queryLatestSlotNo
+        if slotNo <= unSlotNo slot
+          then
+            liftIO . logInfo trce $
+                mconcat
+                  [ "Byron: No rollback required: db tip slot is ", textShow slotNo
+                  , ", ledger tip slot is ", textShow (unSlotNo slot)
+                  ]
+          else do
+            liftIO . logInfo trce $
+                mconcat
+                  [ "Byron: Rollbacking to slot ", textShow (unSlotNo slot)
+                  , ", hash ", Byron.renderAbstractHash hash
+                  ]
+            void . lift $ DB.deleteCascadeSlotNo slotNo
```

Comment:
I think this isn't right yet and can still be a lot simpler.

The `slot` and `hash` we are given is the point to roll back to. So we want to delete blocks _after_ that point, keeping that point at the most recent point.

This means we never need to check `slotNo <= unSlotNo slot` because it's an unnecessary special case. If we're told to roll back to our current tip point then by definition delete blocks _after_ that point is a no-op.

But looking at `DB.deleteCascadeSlotNo slotNo`, it does not delete blocks after the given slot. It deletes blocks in _that_ slot, and _only_ that slot. That cannot be right because it will only delete the current tip block, but we have to be able to cope with rollbacks of arbitrary depth.

So this should be unconditional: just delete _all_ blocks that have a _strictly_ higher slot number than the slot of the rollback point. As an assertion / paranoia check, we could check that the (slot,hash) pair do indeed exist in the DB, so that the rollback makes sense.

There's no special case needed for the "No rollback required" case. That comes out naturally.